### PR TITLE
Fix unpacking zero byte files from NPM

### PR DESCRIPTION
**Pull request by FrederikBolding**
Fixes https://github.com/MetaMask/snaps/issues/1704

This is a fun one. If an NPM package included an empty file of length zero bytes, `concat-stream` would return it as an empty array instead of an `Uint8Array`. This meant that `data.byteLength` would be `undefined` which in turn would break our running counter of total tarball size, resulting in an error that the NPM tarball was too large:
```
number + undefined == NaN
NaN < TARBALL_SIZE_SAFETY_LIMIT == false
```

`concat-stream` supports providing an encoding argument that will convert any output to a given encoding and so the fix is simply providing the expected output type of `Uint8Array` as the specified encoding.

Previously `concat-stream` would try to infer the encoding by looking at the input, but the types did not take that into account causing this unexpected behaviour. 

https://www.npmjs.com/package/concat-stream

### CODE REVIEW

File: packages/snaps-controllers/src/snaps/location/npm.test.ts [hunk 1]
```diff
@@ -66,11 +66,11 @@ describe('NpmLocation', () => {
     const manifest = await location.manifest();
     const sourceCode = (
       await location.fetch(manifest.result.source.location.npm.filePath)
-    ).value.toString();
+    ).toString();
     assert(manifest.result.source.location.npm.iconPath);
     const svgIcon = (
       await location.fetch(manifest.result.source.location.npm.iconPath)
-    ).value.toString();
+    ).toString();
```
Comment (Mrtenz) [inline packages/snaps-controllers/src/snaps/location/npm.test.ts:73]: Why are these changes necessary?

Comment (FrederikBolding) [reply packages/snaps-controllers/src/snaps/location/npm.test.ts:73]: Because we are forcing everything to be `Uint8Array` now and `Uint8Array.toString()` does not encode by default.

My guess is that this probably was a Buffer or something else under test and so it was encoded correctly before, but isn't now that the `value` is guaranteed to be a `Uint8Array`.

Our virtual file implementation has a nicer implementation that handles encoding, so seemed simple to just use that.
